ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -778,6 +778,7 @@ public void onFollowerSuccessAppendEntries(FollowerInfo follower) {
     } else {
       eventQueue.submit(checkStagingEvent);
     }
+    server.getTransferLeadership().onFollowerAppendEntriesReply(follower);
```

Comment:
I wonder if we can "remember" the transferee follower so that we don't have to call it for all the other followers.

Reply:
We could remember transferee, but then we should notify leaderStateImpl when transferee is changed.

Currently we are skipping other followers in `onFollowerAppendEntriesReply()` by this if statement:

```java
    if (follower.getPeer().getId().equals(transferee)) {
```

Reply:
We should use `follower.getId()` instead of `follower.getPeer().getId()` as described in the javadoc.
```java
//FollowerInfo
  /**
   * Return this follower's {@link RaftPeer}.
   * To obtain the {@link RaftPeerId}, use {@link #getId()} which is more efficient than this method.
   *
   * @return this follower's peer info.
   */
  RaftPeer getPeer();
```